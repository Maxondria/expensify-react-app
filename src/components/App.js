import React from "react";
import AppRouter from "../router/AppRouter";
import Header from "./Header";
import { createBrowserHistory } from "history";
import { Router } from "react-router-dom";

export const history = createBrowserHistory();

const App = () => (
  <Router history={history}>
    <Header />
    <AppRouter />
  </Router>
);

export default App;
