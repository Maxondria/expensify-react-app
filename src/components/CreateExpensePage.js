import React from "react";

const CreateExpensePage = () => <h1>Create Expense Form</h1>;

export default CreateExpensePage;
