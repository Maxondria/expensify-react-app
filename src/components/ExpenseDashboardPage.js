import React from "react";

const ExpenseDashboardPage = () => <h1>Expense Dashboard</h1>;

export default ExpenseDashboardPage;
