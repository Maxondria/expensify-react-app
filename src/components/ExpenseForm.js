import React from "react";
import moment from "moment";
import "react-dates/initialize";
import "react-dates/lib/css/_datepicker.css";
import { SingleDatePicker } from "react-dates";

export default class ExpenseForm extends React.Component {
  state = {
    amount: "",
    description: "",
    note: "",
    calenderFocused: false,
    date: moment()
  };

  OnChange = ({ target: { name, value } }) => {
    this.setState(prevState => ({
      ...prevState,
      [name]:
        name === "amount"
          ? value.match(/^\d+(\.\d{0,2})?$/)
            ? value
            : ""
          : value
    }));
  };

  OnSave = e => {
    e.preventDefault();
    console.log(this.state);
  };

  OnFocusChange = ({ focused }) => this.setState({ calenderFocused: focused });

  OnDateChange = date => this.setState({ date });

  render() {
    return (
      <div>
        <form>
          <input
            type="text"
            name="description"
            value={this.state.description}
            placeholder="Description"
            onChange={this.OnChange}
            autoFocus
          />
          <input
            type="text"
            name="amount"
            value={this.state.amount}
            onChange={this.OnChange}
            placeholder="Amount"
          />
           {/* eslint-disable react/prop-types */}
          <SingleDatePicker
            date={this.state.date}
            onDateChange={this.OnDateChange}
            focused={this.state.calenderFocused}
            onFocusChange={this.OnFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
            id="CreatedAt"
          />
          <textarea
            cols="20"
            rows="5"
            name="note"
            placeholder="Add A Note"
            onChange={this.OnChange}
            value={this.state.note}
          />
          <button onClick={this.OnSave}>Save Expense</button>
        </form>
      </div>
    );
  }
}
