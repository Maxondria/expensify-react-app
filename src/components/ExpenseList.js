import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";

const ExpenseList = props => (
  <div>
    <h1>This is my Expenses List</h1>
    {props.expenses.map(expense => (
      <ExpenseListItem {...expense} />
    ))}
  </div>
);

const mapStateToProps = ({ expenses, filters }) => ({
  expenses,
  filters
});

export default connect(mapStateToProps)(ExpenseList);
