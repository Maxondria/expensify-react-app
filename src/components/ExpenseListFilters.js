import React from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  sortByAmount,
  sortByDate
} from "../redux/actions/filters";

const ExpenseListFilters = props => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={e => props.setTextFilter(e.target.value)}
    />

    <select
      name="sort_by"
      id="sort_by"
      value={props.filters.sortBy}
      onChange={e => {
        return e.target.value === "date"
          ? props.sortByDate()
          : props.sortByAmount();
      }}
    >
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = ({ filters }) => ({
  filters
});

const mapDispatchToProps = {
  setTextFilter,
  sortByAmount,
  sortByDate
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExpenseListFilters);
