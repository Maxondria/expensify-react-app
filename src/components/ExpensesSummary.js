import React from "react";
import numeral from "numeral";
import { connect } from "react-redux";
import expensesTotalSelector from "../redux/selectors/expense-total";
import expensesSelector from "../redux/selectors/expenses";

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedTotal = numeral(expensesTotal).format("$0,0.00");

  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totaling {formattedTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = ({ expenses, filters }) => {
  const visibleExpenses = expensesSelector(expenses, filters);
  return {
    expensesTotal: expensesTotalSelector(visibleExpenses),
    expenseCount: visibleExpenses.length
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
