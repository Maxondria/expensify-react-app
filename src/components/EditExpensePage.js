import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, startRemoveExpense } from "../redux/actions/expenses";

export class EditExpensePage extends React.Component {
  id = this.props.match.params.id;

  onSubmit = ({ note, description, amount, date: createdAt }) => {
    this.props.editExpense(this.id, {
      note,
      description,
      amount,
      createdAt
    });
    this.props.history.push("/");
  };

  onRemoveExpense = () =>
    this.props.startRemoveExpense(this.id) && this.props.history.push("/");

  render() {
    return (
      <div>
        {this.props.expense.length === 0 ? (
          <p>Expense Not Found</p>
        ) : (
          <>
            <ExpenseForm
              expense={this.props.expense}
              onSubmit={this.onSubmit}
            />
            <button onClick={this.onRemoveExpense}>Remove</button>
          </>
        )}
      </div>
    );
  }
}

const mapStateToProps = ({ expenses }, props) => ({
  expense: expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchToProps = {
  editExpense,
  startRemoveExpense
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
