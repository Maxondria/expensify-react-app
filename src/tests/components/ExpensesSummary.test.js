import React from "react";
import { ExpenseSummary } from "../../components/ExpensesSummary";
import { shallow } from "enzyme";

const singleExpense = { expenseCount: 1, expensesTotal: 90 };
const multipleExpenses = { expenseCount: 10, expensesTotal: 900 };

it("should render ExpensesSummary correctly when a single expense is provided", function() {
  const wrapper = shallow(<ExpenseSummary {...singleExpense} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("h1").text()).toBe("Viewing 1 expense totaling $90.00");
});

it("should render ExpensesSummary correctly when a single expense is provided", function() {
  const wrapper = shallow(<ExpenseSummary {...multipleExpenses} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("h1").text()).toBe(
    "Viewing 10 expenses totaling $900.00"
  );
});
