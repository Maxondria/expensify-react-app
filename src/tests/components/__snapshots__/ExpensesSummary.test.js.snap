// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly when a single expense is provided 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $90.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly when a single expense is provided 2`] = `
<div>
  <h1>
    Viewing 
    10
     
    expenses
     totaling 
    $900.00
  </h1>
</div>
`;
